sdk/python/pvsite_datamodel/write/generation.py

```diff
@@ -70,4 +83,4 @@ def insert_generation_values(
 
         generation_sqls.append(generation)
 
-    _upsert(session, GenerationSQL, generation_sqls)
+    _upsert_do_nothing(session, GenerationSQL, generation_sqls)
```

Comment:
We should remove the `_upsert` function then, I think this is the only place it is used.

Reply:
True, i guess its in the git history if we every need it